MerlinAU.sh

```diff
@@ -4523,63 +4712,200 @@ _CapitalizeFirstChar_()
+   if [ "$numDays1" -lt "$numDays2" ]
+   then return 0
+   else return 1
+   fi
+}
+
+##-------------------------------------##
+## Added by Martinski W. [2024-Nov-24] ##
+##-------------------------------------##
+#-----------------------------------------------#
+# Allow ONLY the valid range [0-6] and some
+# specific intervals [ *  */2  */3 ] for the
+# the purposes of doing F/W Updates.
+#-----------------------------------------------#
+_ValidateCronScheduleDAYofWEEK_()
+{
+   if [ $# -eq 0 ] || [ -z "$1" ] ; then return 1 ; fi
+   if [ "$1" = "*" ] || echo "$1" | grep -qE "^[*]/(2|3)$"
+   then return 0 ; fi
+   if echo "$1" | grep -qE "^${CRON_DAYofWEEK_RegEx}$"
+   then
+       if echo "$1" | grep -q '-'
+       then
+           if _ValidateCronNumOrderDAYW_ "$1"
+           then return 0 ; else return 1 ; fi
+       fi
+       return 0
+   fi
+   return 1
+}
+
+##-------------------------------------##
+## Added by Martinski W. [2024-Nov-24] ##
+##-------------------------------------##
+#--------------------------------------------------#
+# Allow ONLY the valid range [1-31] and some
+# intervals [ *  */[2-9]  */10  */12  */15 ]
+# for the purposes of doing F/W Updates.
+#--------------------------------------------------#
+_ValidateCronScheduleDAYofMONTH_()
+{
+   if [ $# -eq 0 ] || [ -z "$1" ] ; then return 1 ; fi
+   if [ "$1" = "*" ] || \
+      echo "$1" | grep -qE "^[*]/([2-9]|10|12|15)$"
+   then return 0 ; fi
+   if echo "$1" | grep -qE "^${CRON_DAYofMONTH_RegEx}$"
+   then
+       if echo "$1" | grep -q '-'
+       then
+           local numDays1  numDays2
+           numDays1="$(echo "$1" | awk -F '-' '{print $1}')"
+           numDays2="$(echo "$1" | awk -F '-' '{print $2}')"
+           if [ "$numDays1" -lt "$numDays2" ]
+           then return 0
+           else return 1
+           fi
+       fi
+       return 0
+   fi
+   return 1
+}
+
+##----------------------------------------##
+## Modified by Martinski W. [2024-Nov-24] ##
+##----------------------------------------##
 _ValidateCronJobSchedule_()
 {
-   local cronSchedsStr
+   local cronSchedStr  cronSchedDAYW  cronSchedDAYM  cronSchedMNTH
+
    if [ $# -eq 0 ] || [ -z "$1" ]
    then
        printf "${REDct}INVALID cron schedule string: [EMPTY].${NOct}\n"
        return 1
    fi
-   cronSchedsStr="$(echo "$1" | awk -F ' ' '{print NF}')"
-   if [ "$cronSchedsStr" -ne 5 ]
+   cronSchedStr="$(echo "$1" | awk -F ' ' '{print NF}')"
+   if [ "$cronSchedStr" -ne 5 ]
    then
        printf "${REDct}INVALID cron schedule string [$1]. Incorrect number of parameters.${NOct}\n"
        return 1
    fi
-   cronSchedsStr="$(echo "$1" | awk -F ' ' '{print $1}')"
-   if ! echo "$cronSchedsStr" | grep -qE "^(${CRON_MINS_RegEx})$"
+   cronSchedStr="$(echo "$1" | awk -F ' ' '{print $1}')"
+   if ! _ValidateCronScheduleMINS_ "$cronSchedStr"
+   then
+       printf "${REDct}INVALID cron value for 'MINUTE' [$cronSchedStr].${NOct}\n"
```

Comment:
I would clarify this, make the requirement clear that it needs to be a full minute now

Reply:
> I would clarify this, make the requirement clear that it needs to be a full minute value now
> 
Good point. I'll review the messaging and modify it for more clarity.

Reply:
@Martinski4GitHub 

This is a change in behavior and if one thing my job as a workstation engineer taught me (even though it wasn't for long) is changes in behaviors need to be clearly communicated to users else they just go based on memory and get confused when it doesn't work like it used too.

I'm fine with documenting the changes to the cron behavior in the Wiki, but for users trying to set it live we should give them a bit of context I think

Reply:
> @Martinski4GitHub
> 
> This is a change in behavior and if one thing my job as a workstation engineer taught me (even though it wasn't for long) is changes in behaviors need to be clearly communicated to users else they just go based on memory and get confused when it doesn't work like it used too. LOOL
> 
> I'm fine with documenting the changes to the cron behavior in the Wiki, but for users trying to set it live we should give them a bit of context here I think
>
>
When you get a chance, take a look at the new error messages getting generated when validation fails. They're more verbose and provide more details about the new restrictions on the cron schedule parameters.

Remember that the cron is intended for a very specific purpose so we don't want users to be able to set up a schedule that is counter-productive to do F/W Updates. For example, before we were allowing users to set a cron job to run "every minute in every hour" which would be the wrong thing to do. The intention now is to "guide" the users to  make more reasonable choices, especially with MINUTE & HOUR parameters.